xcp/pci.py

```diff
@@ -326,6 +328,8 @@ def pci_sbdfi_to_nic(sbdfi, nics):
     @raises: Exception if SBDF[index] is not found
     """
     match = VALID_SBDFI.match(sbdfi)
+    if TYPE_CHECKING:
+        assert match
```

Comment:
Maybe it's because I'm not used to seeing TYPE_CHECKING, but I'd prefer if the match happened inside a try ... catch block, do you think this is reasonable for type checkers as well?

```python
try:
    match = VALID_SBDFI.match(sbdfi)
except:
    raise Exception("...")
assert match
```

Reply:
@psafont

- Your suggestion is slightly off because .match just returns None in case of no match
- The raised Exception is the AttributeError, which would be raised afterwards.

To not change behavior for any callers:

Before implementing something along these lines, I now pushed fixes for the testcases to verify the Exception type and string exactly.

I CI check 1st commit to let so the CI confirms that the new test is checking picking up the currently raised exception for all possible causes.

In a 2nd commit, I then pushed the explicit raise of the current Exception and the assert.